### ISSUE TITLE
Dasboard - use secrets data from test-with-executions/TESTNAME instead of list

### ISSUE THREAD

**Issue by exu** [kubeshop/testkube#1896]
In the context of performance issues:

I’ve had a session with Alex today and we figured out that the issue is about the whole architecture of listing our executions

Currently, we’re storing all data inside a list which will not scale
In terms of the amount of data

Each test on the list with secrets data needs to call for secrets so if there is more secrets used endpoint will get slower.

I think we should start asking on the front end about particular resources (not have all on the list) e.g. when we’re loading Tests - call `/v1/tests-with-executions/testname` for a particular test and on the backend side.

So to sum up: I would love to see the new design follow this simple rule - to make a detailed call to the particular resource instead of having everything on lists - as it’ll not scale at all
on backend side we need to remove calls to secrets for list.

**Comment by exu** [comment]
Ping backend guys after complete this one to do #1897
